sources/web/datalab/settings.ts

```diff
@@ -186,15 +205,30 @@ export function updateUserSetting(userId: string, key: string, value: string, as
  * @param response the outgoing http response.
  */
 function requestHandler(request: http.ServerRequest, response: http.ServerResponse): void {
-  var userId = userManager.getUserId(request);
-  if ('POST' == request.method) {
-    postSettingsHandler(userId, request, response);
+  if (request.url.indexOf('/api/settings') === 0) {
```

Comment:
Both as of this update. It handles calls to `/api/settings` (app settings) as well as `/_settings` (user settings).